libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -33,6 +36,12 @@ export class BookSearchComponent implements OnInit, OnDestroy {
 
   ngOnInit(): void {
     this.books = this.store.select(getAllBooks);
+    this.searchTextChanged
+      .pipe(debounceTime(500), distinctUntilChanged())
+      .subscribe((searchstring) => {
+        this.searchForm.controls.term.setValue(searchstring);
+        this.searchBooks();
+      });
```

Comment:
To provide instant search results as the user is typing. API request no more than one request every 500 ms